src/store/index.ts

```diff
@@ -0,0 +1,31 @@
+import { applyMiddleware, compose, createStore } from "redux";
+import createSagaMiddleware from "redux-saga";
+import createReducer from "./reducer";
+
+
+declare global {
+  interface Window {
+    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
+  }
+}
+
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+
+
+const sagaMiddleware = createSagaMiddleware();
+const enhancer = composeEnhancers(applyMiddleware(sagaMiddleware));
+
+const store: any = createStore(createReducer({}), enhancer);
+
+store.asyncReducers = {};
+
+export const injectReducer = (key: string, reducer: object) => {
+  if (store.asyncReducers[key]) {
```

Comment:
What's the purpose of this check here? I don't see it in the example function here: https://redux.js.org/recipes/code-splitting/#defining-an-injectreducer-function

Also, in the example above, `injectReducer` is a property of `store` rather than a global function. Is there a reason for that?